File: 1. Easy/232. Implement Queue using Stacks/solution.js
var MyQueue = function () {
    this.instack = [];
    this.outstack = [];
};

MyQueue.prototype.push = function (x) {
    while (this.outstack.length > 0)
        this.instack.push(this.outstack.pop());
    this.instack.push(x);
};

MyQueue.prototype.pop = function () {
    while (this.instack.length > 0)
        this.outstack.push(this.instack.pop());
    return this.outstack.pop();
};

MyQueue.prototype.peek = function () {
    while (this.instack.length > 0)
        this.outstack.push(this.instack.pop());
    return this.outstack[this.outstack.length - 1];
};

MyQueue.prototype.empty = function () {
    return this.instack.length === 0 && this.outstack.length === 0;
};

File: 1. Easy/232. Implement Queue using Stacks/Solution.java
class MyQueue {
    private final Stack<Integer> in;
    private final Stack<Integer> out;

    public MyQueue() {
        this.in = new Stack<>();
        this.out = new Stack<>();
    }

    public void push(int x) {
        while (!out.isEmpty())
            in.push(out.pop());

        in.push(x);
    }

    public int pop() {
        while (!in.isEmpty())
            out.push(in.pop());

        return out.pop();
    }

    public int peek() {
        while (!in.isEmpty())
            out.push(in.pop());

        return out.peek();
    }

    public boolean empty() {
        return in.isEmpty() && out.isEmpty();
    }
}